lib/vsc/install/ci.py

```diff
@@ -0,0 +1,163 @@
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# the Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/hpcugent/vsc-install
+#
+# vsc-install is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Library General Public License as
+# published by the Free Software Foundation, either version 2 of
+# the License, or (at your option) any later version.
+#
+# vsc-install is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Library General Public License for more details.
+#
+# You should have received a copy of the GNU Library General Public License
+# along with vsc-install. If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Generate configuration files for running CI tests.
+
+Run with: python -m vsc.install.ci
+
+@author: Kenneth Hoste (Ghent University)
+"""
+import logging
+import os
+
+
+JENKINSFILE = 'Jenkinsfile'
+TOX_INI = 'tox.ini'
+
+logging.basicConfig(format="%(message)s", level=logging.INFO)
+
+
+def write_file(path, txt):
+    """Write specified contents to specified path."""
+    try:
+        with open(path, 'w') as handle:
+            handle.write(txt)
+        logging.info("Wrote %s", path)
+    except (IOError, OSError) as err:
+        raise IOError("Failed to write %s in %s: %s" % (path, os.getcwd(), err))
+
+
+def gen_tox_ini(install_vsc_install):
+    """
+    Generate tox.ini configuration file for tox
+    see also https://tox.readthedocs.io/en/latest/config.html
+    """
+    logging.info('[%s]', TOX_INI)
+
+    header = [
+        "%s: configuration file for tox" % TOX_INI,
+        "This file was automatically generated using 'python -m vsc.install.ci'",
+        "DO NOT EDIT MANUALLY",
+    ]
+    header = ['# ' + l for l in header]
+
+    py3_env = 'py36'
+    envs = ['py27', py3_env]
+
+    lines = header + [
+        '',
+        "[tox]",
+        "envlist = %s" % ','.join(envs),
+        # instruct tox not to run sdist prior to installing the package in the tox environment
+        # (setup.py requires vsc-install, which is not installed yet when 'python setup.py sdist' is run)
+        "skipsdist = true",
+        # ignore failures due to missing Python version
+        # python2.7 must always be available though, see Jenkinsfile
+        "skip_missing_interpreters = true",
+        '',
+        '[testenv]',
+    ]
+
+    if install_vsc_install:
+        lines.extend([
+            # use easy_install rather than pip to install vsc-install dependency
+            # (vsc-* packages may not work when installed with pip due to use of namespace package vsc.*)
```

Comment:
what is this about? why would one need pip or anything like that?

Reply:
ok, think i understand. it 's a bit tricky to bootstrap possibly, but other than that, should be ok. i sortoffrogit how this used to work before...

Reply:
`pip` is the recommended tool to install Python packages (and I think `setuptools`/`easy_install` is even deprecated), but we can't use it because of the way we do the `vsc.*` namespace currently (which is not fully supported by `pip` for some bizarre reason)